javatests/arcs/android/host/AndroidEntityHandleManagerTest.kt

```diff
@@ -69,37 +71,24 @@ class AndroidEntityHandleManagerTest {
     @Before
     fun setUp() {
         app = ApplicationProvider.getApplicationContext()
-        app.setTheme(R.style.Theme_AppCompat);
+        lifecycle = LifecycleRegistry(this).apply {
+            setCurrentState(Lifecycle.State.CREATED)
+            setCurrentState(Lifecycle.State.STARTED)
+            setCurrentState(Lifecycle.State.RESUMED)
+        }
 
         // Initialize WorkManager for instrumentation tests.
         WorkManagerTestInitHelper.initializeTestWorkManager(app)
 
         handleHolder = TestParticleHandles()
-    }
 
-    fun handleManagerTest(
-        block: suspend TestCoroutineScope.(EntityHandleManager) -> Unit
-    ) = runBlockingTest {
-        val scenario = ActivityScenario.launch(TestActivity::class.java)
-
-        scenario.moveToState(Lifecycle.State.STARTED)
-
-        val activityJob = launch {
-            scenario.onActivity { activity ->
-                val hf = AndroidHandleManager(
-                    lifecycle = activity.lifecycle, context = activity,
-                    connectionFactory = DefaultConnectionFactory(
-                        activity, TestBindingDelegate(app), coroutineContext
-                    ),
-                    coroutineContext = coroutineContext
-                )
-                runBlocking {
-                    this@runBlockingTest.block(EntityHandleManager(hf))
-                }
-                scenario.close()
-            }
-        }
-        activityJob.join()
```

Comment:
I'm wondering if this is going to cause problems, but I admit I'm an android n00b here. What I worry about is that if we startup 2 Android services, are they running in different threads by Robolectric, or is this single threaded?

If startService/startAcvtivity always results in a single threaded dispatch it'll work, but if it's multithreaded, I'm wondering if the test method will return before jobs that are run inside of a Service finish, which will cause those "Job Not Completion" exceptions we see sometimes?

Reply:
It turns out there's an issue with `runBlockingTests` where even if jobs are properly completed, but on a different thread, `runBlockingTest` will report "This job has not completed", even after joining all child jobs.